File: app/src/main/java/com/ndipatri/iot/googleproximity/sample/GoogleProximityApp.java
package com.ndipatri.iot.googleproximity.sample;

import android.app.Application;
import android.util.Log;

import com.ndipatri.iot.googleproximity.GoogleProximity;

public class GoogleProximityApp extends Application {

    private static final String TAG = GoogleProximityApp.class.getSimpleName();

    private static GoogleProximityApp instance;

    public GoogleProximityApp() {
        GoogleProximityApp.instance = this;
    }

    public static GoogleProximityApp getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        Log.d(TAG, "Starting app ...");

        GoogleProximity.initialize(this, true);
    }
}

File: app/src/main/java/com/ndipatri/iot/googleproximity/sample/GoogleProximityApp.kt
package com.ndipatri.iot.googleproximity.sample

import android.app.Application
import android.util.Log

import com.ndipatri.iot.googleproximity.GoogleProximity

class GoogleProximityApp : Application() {
    init {
        GoogleProximityApp.instance = this
    }

    override fun onCreate() {
        super.onCreate()

        Log.d(TAG, "Starting app ...")

        GoogleProximity.initialize(this, true)
    }

    companion object {

        private val TAG = GoogleProximityApp::class.java!!.getSimpleName()

        var instance: GoogleProximityApp? = null
            private set
    }
}
